dts/arm/nuvoton/npck/npck.dtsi

```diff
@@ -376,7 +376,7 @@
 		};
 
 		twd0: watchdog@400d8000 {
-			compatible = "nuvoton,npcx-watchdog";
+			compatible = "nuvoton,npcx-watchdog", "nuvoton,npcx-watchdog-npckn-v1";
 			reg = <0x400d8000 0x2000>;
 			t0-out = <&wui_t0out>;
```

Comment:
There are two ways to do this in zephyr

a. Add a separate compatible, like "nuvoton,npcx-watchdog-npckn-v1", and use that as the primary and only compatible for the hardware. We don't have loadable drivers to choose from at runtime in order of specificity, the one driver includes is the one you get, so only the one, specific compatible you will build a driver for should be present.

b. add the extended feature as a flag, which seems like the way easiest solution here. add a boolean like `is-extended;` or `supports-extended;` or something to the binding for "nuvoton,npcx-watchdog", and in the driver, check for the flag, store it in config, and check for it in the driver, no new Kconfig required :)

Reply:
yeah I second this, property seems the easier to maintain, but see my comment above, if this matches with some specific soc families it may be even simpler to use one of those options, ST does it all over the place (and to be perfectly honest the result is not pretty, but it works for them)

Reply:
Yeah the suggestion is to add a boolean property instead of a new compatible, should be a way less intrusive change, especially if you have more variations in the future.

Reply:
We also saw the similar usage in [lpuart1](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/st/l4/stm32l4.dtsi#L233) node of stm32 that assigns two compatible strings `compatible = "st,stm32-lpuart", "st,stm32-uart";`  in the `compatible` property and use the string `st_stm32_lpuart` to guard its specific codes in the [driver](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/uart_stm32.c#L59).